include/cassandra.h

```diff
@@ -4720,6 +4829,122 @@ cass_result_has_more_pages(const CassResult* result);
 
 /***********************************************************************************
  *
+ * Error result
+ *
+ ***********************************************************************************/
+
+/**
+ * Frees an error result instance.
+ *
+ * @public @memberof CassErrorResult
+ *
+ * @param[in] result
+ */
+CASS_EXPORT void
+cass_error_result_free(const CassErrorResult* error_result);
+
+/**
+ * Gets error code for the error result. This error code will always
+ * have an server error source.
+ *
+ * @public @memberof CassErrorResult
+ *
+ * @param[in] error_result
+ * @return The server error code
+ */
+CASS_EXPORT CassError
+cass_error_result_code(const CassErrorResult* error_result);
+
+/**
+ * Gets consistency that triggered the error result of the
+ * following types:
+ *
+ * <ul>
+ *   <li>CASS_ERROR_SERVER_READ_TIMEOUT</li>
+ *   <li>CASS_ERROR_SERVER_WRITE_TIMEOUT</li>
+ *   <li>CASS_ERROR_SERVER_UNAVAILABLE</li>
+ * </ul>
+ *
+ * @public @memberof CassErrorResult
+ *
+ * @param[in] error_result
+ * @return The consistency that triggered the error for a read timeout,
+ * write timeout or an unavailable error result. Undefined for other
+ * error result types.
+ */
+CASS_EXPORT CassConsistency
+cass_error_result_consistency(const CassErrorResult* error_result);
+
+/**
+ * Gets the acutal number of recieved responses, recieved acknowlegements
+ * or alive nodes for following error result types, respectively:
+ *
+ * <ul>
+ *   <li>CASS_ERROR_SERVER_READ_TIMEOUT</li>
+ *   <li>CASS_ERROR_SERVER_WRITE_TIMEOUT</li>
+ *   <li>CASS_ERROR_SERVER_UNAVAILABLE</li>
+ * </ul>
+ *
+ * @public @memberof CassErrorResult
+ *
+ * @param[in] error_result
+ * @return The actual received responses for a read timeout, actual recieved
+ * acknowlegements for a write timeout or actual alive nodes for a unavailable
+ * error. Undefined for other error result types.
+ */
+CASS_EXPORT cass_int32_t
+cass_error_result_actual(const CassErrorResult* error_result);
```

Comment:
The problem is it different for every error type. It's not always "received responses" it could be "number of alive nodes" or "number of acknowledgements". I was trying to avoid adding a new function for every type because it's really the same-ish value.